src/pathfinding/grid_pathfinder.jl

```diff
@@ -0,0 +1,362 @@
+
+            if new_g_cost < nbor_cell.g
+                parent[nbor] = cur
+                grid[nbor] = GridCell(
+                    new_g_cost,
+                    delta_cost(pathfinder, nbor, to),
+                    pathfinder.admissibility,
+                )
+                if haskey(open_list_handles, nbor)
+                    update!(open_list, open_list_handles[nbor], (grid[nbor].f, nbor))
+                else
+                    open_list_handles[nbor] = push!(open_list, (grid[nbor].f, nbor))
+                end
+            end
+        end
+    end
+
+    agent_path = Path{D}()
+    cur = to
+    while true
+        haskey(parent, cur) || break
+        pushfirst!(agent_path, cur)
+        cur = parent[cur]
+    end
+    return agent_path
+end
+
+@inline get_neighbors(cur, pathfinder::AStar{D,true}) where {D} =
+    (mod1.(cur .+ β.I, pathfinder.grid_dims) for β in pathfinder.neighborhood)
+@inline get_neighbors(cur, pathfinder::AStar{D,false}) where {D} =
+    (cur .+ β.I for β in pathfinder.neighborhood)
+@inline inbounds(n, pathfinder, closed) =
+    all(1 .<= n .<= pathfinder.grid_dims) && pathfinder.walkable[n...] && n ∉ closed
+
+"""
+    set_target!(agent::A, target::NTuple{D,Int}, model::ABM{<:GridSpace{D,P,<:AStar{D}},A})
+This calculates and stores the shortest path to move the agent from its current position to `target`
+using [`find_path`](@ref).
+"""
+function set_target!(
+    agent::A,
+    target::Dims{D},
+    model::ABM{<:GridSpace{D,P,<:AStar{D}},A},
+) where {D,P,A<:AbstractAgent}
+    model.space.pathfinder.agent_paths[agent.id] = find_path(model.space.pathfinder, agent.pos, target)
+end
+
+"""
+    is_stationary(agent, model::ABM{<:GridSpace{D,P,<:AStar{D}},A})
+Return `true` if agent has reached it's target destination, or no path has been set for it.
+"""
+is_stationary(
+    agent::A,
+    model::ABM{<:GridSpace{D,P,<:AStar{D}},A},
+) where {D,P,A<:AbstractAgent} = isempty(agent.id, model.space.pathfinder)
+
+Base.isempty(id::Int, pathfinder::AStar) =
+    !haskey(pathfinder.agent_paths, id) || isempty(pathfinder.agent_paths[id])
+
+"""
+    heightmap(model::ABM{<:GridSpace{D,P,<:AStar{D}}})
+Return the heightmap of the pathfinder if the [`HeightMap`](@ref) metric is in use,
+`nothing` otherwise.
+"""
+function heightmap(model::ABM{<:GridSpace{D,P,<:AStar{D}}}) where {D,P}
+    if model.space.pathfinder.cost_metric isa HeightMap
+        return model.space.pathfinder.cost_metric.hmap
+    else
+        return nothing
+    end
+end
+
+"""
+    walkmap(model::ABM{<:GridSpace{D,P,<:AStar{D}}})
+Return the walkable map of the pathfinder
+"""
+walkmap(model::ABM{<:GridSpace{D,P,<:AStar{D}}}) where {D,P} = model.space.pathfinder.walkable
+
+"""
+    move_agent!(agent::A, model::ABM{<:GridSpace{D,P,<:AStar{D}},A})
+Moves the agent along the path to its target set by [`set_target!`](@ref). If the agent does
+not have a precalculated path, or the path is empty, the agent does not move.
+"""
```

Comment:
Ah... unfortunately, as much as I like to keep the total number of function names to a minimum, here we cannot do this. The function `move_agent!(agent, space)` moves the agent to a random location in a discrete space. So we could replace the name with `move_along_path!` for example. 

I am afraid that if we simply over-write the existing `move_agent!(agent, space)` will be too confusing.

Reply:
Also, if we have `move_along_path!` we can expand this documentation string in the "Path-finding" section of the API, which is more natural.

Reply:
@Libbum in essence `move_along_path!` is the same function as the OSM space one, right? Perhaps in the future we can consider renaming the OSM function to have the same name.

Reply:
A good suggestion. Let's do that.